src/pages/preferences/preferencesTabs/publicKeysTab/EditModal.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useEffect } from "react";
+import { Modal } from "components/Modal";
+import { useMutation } from "@apollo/react-hooks";
+import styled from "@emotion/styled";
+import Button, { Variant } from "@leafygreen-ui/button";
+import {
+  GetMyPublicKeysQuery,
+  UpdatePublicKeyMutation,
+  UpdatePublicKeyMutationVariables,
+  GetMyPublicKeysQueryVariables,
+  CreatePublicKeyMutation,
+  CreatePublicKeyMutationVariables,
+} from "gql/generated/types";
+import { GET_MY_PUBLIC_KEYS } from "gql/queries";
+import { UPDATE_PUBLIC_KEY } from "gql/mutations/update-public-key";
+import { useBannerDispatchContext } from "context/banners";
+import { Input } from "antd";
+import { CREATE_PUBLIC_KEY } from "gql/mutations/create-public-key";
+import { InputLabel, ErrorMessage } from "components/styles";
+import { v4 as uuid } from "uuid";
+
+const { TextArea } = Input;
+
+export interface EditModalPropsState {
+  replaceKeyName?: string;
+  initialKeyName?: string;
+  initialKeyValue?: string;
+  visible: boolean;
+  existingKeys?: {
+    name: string;
+    key?: string;
+  }[];
+}
+
+interface EditModalProps extends EditModalPropsState {
+  onCancel: () => void;
+}
+
+export const EditModal: React.FC<EditModalProps> = ({
+  replaceKeyName,
+  initialKeyName,
+  initialKeyValue,
+  existingKeys,
+  visible,
+  onCancel,
```

Comment:
I'm not convinced this is the best approach.
 I dont think we need to pass in all this information from the parent into this component. Doing it this way seems like we are managing this modals state in two separate places. I think it would be better to just pass in a publicKey as a prop. 
IF a publicKey is passed in then this would be an edit modal otherwise it would be a create new key modal. You can handle keeping track of the initial passed in values from within the modal itself.

Reply:
hm, this is a good point. we can probably, store replaceKeyName inside the EditModal, initialKeyName and initialKeyValue can be combined into 1 prop and existingKeys can be read from EditModal.
After those changes the props could be `visible`, `onCancel`, and `publicKey?`